src/contexts/Api/index.tsx

```diff
@@ -66,36 +71,14 @@ export const APIProvider = ({ children }: { children: React.ReactNode }) => {
     !!localStorage.getItem('isLightClient')
   );
 
-  // initial connection
   useEffect(() => {
-    if (!provider) {
-      const _network: NetworkName = localStorage.getItem(
-        'network'
-      ) as NetworkName;
-      connect(_network, isLightClient);
-    }
-  });
+    connect(network.name, isLightClient);
+  }, []);
 
-  // provider event handlers
-  useEffect(() => {
-    if (provider !== null) {
-      provider.on('connected', () => {
-        setConnectionStatus(ConnectionStatus.Connected);
-      });
-      provider.on('error', () => {
-        setConnectionStatus(ConnectionStatus.Disconnected);
-      });
-      connectedCallback(provider);
-    }
-  }, [provider]);
```

Comment:
This contributes to an uncessary `useEffect`-waterfall-logic - this logic should be right [where the `provider` is initialized](https://github.com/Cardinal-Cryptography/aleph-zero-dashboard/pull/37/files#diff-de701a1090eab6a6ce998b10578ba8d16bd3d81a57a7caf3ea8d1d14141ab0a8R172-R179).